src/sync/mod.rs

```diff
@@ -4,6 +4,149 @@
 //!
 //! [`std::sync`]: https://doc.rust-lang.org/std/sync/index.html
 //!
+//! ## The need for synchronization
```

Comment:
I think it's worth adding a sentence right on top on _why_ `async-std` has their own structures (scheduler-awareness). It's a bit hidden below.

Reply:
"async-std's sync primitives are scheduler-aware, making it possible to `.await` their operations - for example the locking of a Mutex"

Reply:
@skade that's an excellent suggestion. Adding it!